reader_concurrency_semaphore.cc

```diff
@@ -320,6 +321,17 @@ class reader_permit::impl
         }
         _timeout = timeout;
     }
+
+    query::max_result_size max_result_size() const {
+        if (_max_result_size) {
+            return *_max_result_size;
+        }
+        return _semaphore.unlimited_query_max_result_size();
```

Comment:
Ahh, I see. This is used as the fall-back in case there is no limit set by the coordinator.